packages/venia-concept/src/RootComponents/Category/category.js

```diff
@@ -1,119 +1,97 @@
-            setPrevPageTotal
-        } = this.props;
+    const [queryResult, queryApi] = useQuery(categoryQuery);
+    const { data, error, loading } = queryResult;
+    const { runQuery, setLoading } = queryApi;
+    const classes = mergeClasses(defaultClasses, props.classes);
 
-        const pageControl = {
-            currentPage: currentPage,
-            setPage: setCurrentPage,
-            updateTotalPages: setPrevPageTotal,
-            totalPages: prevPageTotal
-        };
+    useEffect(() => {
+        setLoading(true);
+        runQuery({
+            variables: {
+                id: Number(id),
+                onServer: false,
+                pageSize: Number(pageSize),
+                currentPage: Number(currentPage)
+            }
+        });
+    }, [id, pageSize, currentPage, runQuery, setLoading]);
 
-        return (
-            <Query
-                query={categoryQuery}
-                variables={{
-                    id: Number(id),
-                    onServer: false,
-                    pageSize: Number(pageSize),
-                    currentPage: Number(currentPage)
-                }}
-            >
-                {({ loading, error, data }) => {
-                    if (error) return <div>Data Fetch Error</div>;
-                    // If our pagination component has mounted, then we have
-                    // a total page count in the store, so we continue to render
-                    // with our last known total
-                    if (loading)
-                        return pageControl.totalPages ? (
-                            <CategoryContent
-                                pageControl={pageControl}
-                                pageSize={pageSize}
-                            />
-                        ) : (
-                            loadingIndicator
-                        );
+    useEffect(() => {
+        window.scrollTo({
+            left: 0,
+            top: 0,
+            behavior: 'smooth'
+        });
+    }, [currentPage]);
 
-                    // TODO: Retrieve the page total from GraphQL when ready
-                    const pageCount =
-                        data.category.products.total_count / pageSize;
-                    const totalPages = Math.ceil(pageCount);
-                    const totalWrapper = {
-                        ...pageControl,
-                        totalPages: totalPages
-                    };
-
-                    return (
-                        <CategoryContent
-                            classes={classes}
-                            pageControl={totalWrapper}
-                            data={data}
-                        />
-                    );
-                }}
-            </Query>
+    if (error) return <div>Data Fetch Error</div>;
+    // If our pagination component has mounted, then we have
+    // a total page count in the store, so we continue to render
+    // with our last known total
+    if (loading || !data)
```

Comment:
This change was odd to me; there seemed to be a state where `loading = false` but `data` was still null. Is this expected?

Reply:
Does it eventually resolve `data` or does it stay `null`?

Reply:
`data` does eventually resolve

Reply:
https://github.com/magento-research/pwa-https://github.com/magento-research/pwa-studio/blob/develop/packages/peregrine/src/hooks/useQueryResult.js#L24

I wonder what happens if you modify that default value to something else.
Maybe this is a bug in `useQueryResult`?

Reply:
Reviewed this with @jimbo, and he confirmed that `useQueryResult` has introduced a new "empty" state that we do need to handle (ideally we should have been handling it already). His Search component re-factor demonstrates the `useQueryResult` states a little better, since it has an "empty" state to handle.
```js
    if (error) {
        message = 'An error occurred while fetching results.';
    } else if (loading) {
        message = 'Fetching results...';
    } else if (!data) {
        message = 'Search for a product';
    } else if (!data.products.items.length) {
        message = 'No results were found.';
    } else {
        message = `${data.products.items.length} items`;
    }
```

Reply:
@tjwiebell Right, thanks for adding that. The empty state exists when you don't have an in-flight request (`loading` is `false`) but you haven't received any responses yet (so `data` is `null`).

The empty state in `Autocomplete` occurs when a user hasn't typed anything yet, since typing is what initiates the request. But here, it occurs when the component is first mounted, since the request is initiated *after* mount, as an effect. That may seem counterintuitive, but a React component isn't allowed to initiate requests before mounting (and for good reasons), so we do briefly have a state where there's neither a request nor a response yet.